tests/standalone_tests/test_celery_app.py

```diff
@@ -0,0 +1,369 @@
+    )
+
+    assert (
+        initial_cel_app
+        != CeleryAppFactory().get_celery_app(RABBITMQ_LOCALNODETMP_ADDR)._celery_app
+    ), (
+        "The celery apps should NOT be the same, the rabbitmq restarted and the broker got corrupted."
+        "https://github.com/celery/celery/issues/6912#issuecomment-1107260087"
+    )
+
+    queue_get_node_info_task_and_assert_response(
+        cel_app_wrapper, get_controller_testing_logger
+    )
+
+    kill_service(localnodetmp_node_service)
+
+
+@pytest.mark.slow
+@pytest.mark.integration
+def test_celery_app_didnt_change_too_many_times_after_parallel_submit_task_when_rabbitmq_restarted(
+    localnodetmp_node_service,
+    reset_celery_app_factory,
+    get_controller_testing_logger,
+):
+    cel_app_wrapper = CeleryAppFactory().get_celery_app(RABBITMQ_LOCALNODETMP_ADDR)
+    initial_cel_app = cel_app_wrapper._celery_app
+
+    concurrent_requests = 20
+
+    # Initialize celery app to open channels
+    queue_get_node_info_task_and_assert_response(
+        cel_app_wrapper, get_controller_testing_logger
+    )
+
+    async_results = [
+        cel_app_wrapper.queue_task(
+            task_signature=get_celery_task_signature("get_node_info"),
+            logger=get_controller_testing_logger,
+            request_id=request_id,
+        )
+        for _ in range(concurrent_requests)
+    ]
+
+    # Restart the rabbitmq
+    remove_localnodetmp_rabbitmq()
+    _create_rabbitmq_container(RABBITMQ_LOCALNODETMP_NAME, RABBITMQ_LOCALNODETMP_PORT)
+
+    if not is_localnodetmp_node_service_ok(localnodetmp_node_service):
+        localnodetmp_node_service = create_localnodetmp_node_service()
+
+    with patch.object(
+        CeleryWrapper,
+        "_instantiate_celery_object",
+        wraps=cel_app_wrapper._instantiate_celery_object,
+    ) as instantiate_celery_app_mock:
+        with ThreadPoolExecutor() as executor:
+            future_task_results = [
+                executor.submit(
+                    cel_app_wrapper.get_result,
+                    async_res,
+                    GET_NODE_INFO_TASK_TIMEOUT,
+                    get_controller_testing_logger,
+                )
+                for async_res in async_results
+            ]
+        for future in future_task_results:
+            # The result could throw an exception or return an actual value depending on
+            # if the result was fetched before the rabbitmq restarted.
+            try:
+                future.result()
+                print("No exception!")
+            except Exception as exc:
+                print(f"Exception {type(exc)}")
+
+    assert 1 <= instantiate_celery_app_mock.call_count < concurrent_requests / 2, (
+        "A new celery app should be created, at least once. "
+        "A new celery app is created upon OSError to reset the app after the rabbitmq restart."
+        "The celery app should not be instantiated as many times as the requests. Only upon OSError."
```

Comment:
you still do not understand exactly what you re trying to check here, why do you check the number of times the mock was called? why does it have to be in [1,concurent_requests/2)??

Reply:
```
        "A new celery app should be created, at least once. "
        "A new celery app is created upon OSError and other errors to reset the app after the rabbitmq restart."
        "The OSError and other celery resetting errors could happen more than once, depending on race conditions."
        "The celery app should not be instantiated as many times as the requests, this would mean that we are not"
        "catching the correct errors. "
        "This formula '1 <= instantiate_celery_app_mock.call_count < concurrent_requests / 2' is the best stable "
        "check that could be achieved, balancing between NOT resetting the celery app each time and checking that"
        "it was actually reset."
```

Added this new comment. I wrote these things so its hard for me to how other people see this.

Reply:
ok so, the explanation here is that nothing is really mocked in the `_instantiate_celery_object` method, the only reason this is patched is to count how many times the method that instantiates a new `Celery` object is called. Apart from that, I think the number of times this will be called is very event dependent and not very deterministic, it seems stable now but could start failing.

Reply:
I am not sure if you want me to add documentation about why it is mocked.
I think its visible from the code where in the patch its wrapped by the same method:
```
    with patch.object(
        CeleryWrapper,
        "_instantiate_celery_object",
        wraps=cel_app_wrapper._instantiate_celery_object,
    ) as instantiate_celery_app_mock:
```
and the only place the mock is used is to get the call count.

If it's not understandable just write me the comment you want me to add.

About the 
```
Apart from that, I think the number of times this will be called is very event dependent and not very deterministic, it seems stable now but could start failing.
``` 
I couldn't find a better way to do it and I don't think there is. I have run it 100 times (literally) to test if it's stable and there were no problems. If it starts failing we can have a look at it then and refactor. Unless you have a better idea of course on how to test this.